docs/source/reference/yaml-spec.rst

```diff
@@ -818,6 +833,57 @@ OR
       store: gcs
       mode: MOUNT
 
+SkyPilot also supports volumes mount.
+
+You can mount network volumes (e.g. GCP persistent disks, etc.) or instance volumes (e.g. local SSD) to the instances in the cluster.
+
+This is supported for GCP only for now.
+
+Note: When :ref:`GCP GPUDirect TCPX <config-yaml-gcp-enable-gpu-direct>` is enabled, the mount path is suggested to be under the `/mnt/disks` directory (e.g., `/mnt/disks/data`). This is because Container-Optimized OS (COS) used for the instances with GPUDirect TCPX enabled has some limitations for the file system. Refer to `GCP documentation <https://cloud.google.com/container-optimized-os/docs/concepts/disks-and-filesystem#working_with_the_file_system>`_ for more details about the filesystem properties of COS.
+
+.. code-block:: yaml
+
+  file_mounts:
+    /mnt/path1: # Path to mount the volume on the instance
+      name: volume-name # Name of an existing volume to mount, if not specified, new volume will be created, optional
+      source: /local/path1 # Source local path, do not set it if no need to sync data from local to volume, if specified, the data will be synced to the /mnt/path1/data directory, optional
+      store: volume # For volume mount
+      persistent: True  # If set to False, the newly created volume will be deleted after cluster is downed, for existing volume, it is always True, optional, default: False
+      config:
+        size: 100 # Size of the volume in GB
+        storage_type: network # Type of the volume, either 'network' or 'instance', optional, default: network
+        disk_tier: best # Tier of the volume, same as `resources.disk_tier`, optional, default: best
+        attach_mode: read_write # Attach mode, either 'read_write' or 'read_only', optional, default: read_write
+
+- Mount with existing volume:
+
+.. code-block:: yaml
+
+  file_mounts:
+    /mnt/path1:
+      name: volume-name
+      store: volume
+
+- Mount with a new network volume:
+
+.. code-block:: yaml
+
+  file_mounts:
+    /mnt/path2:
+      store: volume
```

Comment:
How is the name work here if a user does not specify the name? Maybe we should avoid the case that name is not specified?

Reply:
If the name is not specified, we will create new volumes, this is annotated in the details section line 838.

Reply:
Added more descriptions

Reply:
> If the name is not specified, we will create new volumes, this is annotated in the details section line 838.

If we create a new volume, the user does not know the name for that volume right? Can we just allow the user to specify the name for that volume

Reply:
Updated